src/Symfony/Component/Messenger/Transport/Schedule/Schedule.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+use Symfony\Component\Clock\ClockInterface;
+
+/**
+ * @internal used to bypass a different evaluation order issue in class constant initializers
+ *
+ * @see https://wiki.php.net/rfc/new_in_initializers#unsupported_positions
+ */
+const _SCHEDULE_FAR_FUTURE_ = new \DateTimeImmutable('3000-01-01');
+
+class Schedule implements \IteratorAggregate
```

Comment:
```suggestion
final class Schedule implements \IteratorAggregate
```

Reply:
Are you sure?
I did it exactly as you suggest. And have changed it after playing with it.

The easiest way to provide a schedule it extend the `Schedule` class, ond overwrite the constructor:

```php
#[AsSchedule('default')]
class MySchedule extends Schedule {
    public function __construct(ClockInterface $clock) {
        parent::__construct(
            $clone,
            new PeriodicalMessage(/*...*/),
            /* others ... */
        );
    }
}
```

That's it! No factories, no implementations of interfaces, no specific configurations...

And it's optional. It's possible to configure a service in any way.